### PR TITLE
#186 :zap: improve runtime speed by not duplicating repetitive python…

### PR DESCRIPTION
**Pull request by d-ryan-ashcraft**
… and poetry validations

### CODE REVIEW

File: habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/PyenvAndPoetrySetup.java [hunk 1]
```diff
@@ -34,6 +34,10 @@ public class PyenvAndPoetrySetup {
      */
     static final String PYTHON_DEFAULT_VERSION_REQUIREMENT = "3.11.4";
 
+    public static boolean alreadyValidatedPoetryInstallation;
+    public static String alreadyValidatedPoetryVersion;
+    public static String priorActivePythonVersionActivated;
```
Comment (epwilkins) [inline habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/PyenvAndPoetrySetup.java]: Q: Is Habushu generally not marked thread-safe and/or do we not have have to worry about any race conditions for these checks?

**Review by epwilkins — APPROVED** [review]
If the concern around parallel builds is unfounded, then LGTM

File: habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/PyenvAndPoetrySetup.java (hunk 1 above)
Comment (d-ryan-ashcraft) [reply habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/PyenvAndPoetrySetup.java]: That's a good question. To be sure, I can add a synchronized block to the update for Poetry. The Python version can be added to a ThreadLocal to ensure consistency across the threads.  

We could also mark it explicitly as not thread-safe, as we rarely see anyone do a multi-threaded build due to the complexity of debugging it.

But it's easy enough to support this, so I'll make the updates.

**Comment by d-ryan-ashcraft** [comment]
> If the concern around parallel builds is unfounded, then LGTM

Updated. If you run it with `-T2` now, you'll see it respects both threads with independent validations for their first runs.

**Review by epwilkins — APPROVED** [review]

**Pull request merged**